protocols/abci/addressing-1174.md

```diff
@@ -0,0 +1,241 @@
+$valid(v, bc_{h-1}, x_p)$ to implement in `PrepareProposal`  and `ProcessProposal`
+may break Requirements 3 to 5.
+
+Consensus _validity_ property is then modified as follows:
+
+- _weak validity_: function $valid(v, bc_{h-1}, x_p)$ has returned _true_ at least once
+  by a correct process for the decided block $v$.
+
+### Eventual Requirements
+
+We now relax the relevant ABCI 1.0 requirements in the following way.
+
+* Requirement 3b [`PrepareProposal`, `ProcessProposal`, eventual coherence]:
+  There exists a time $ts_{h}$ for every height $h$ such that,
+  for any two correct processes $p$ and $q$ and any round $r_p$ in height $h$ starting after $ts_{h}$,
+  if $q$'s CometBFT calls `RequestProcessProposal` on $u_p$,
+  $q$'s application returns _Accept_ in `ResponseProcessProposal`.
+
+* The determinism-related requirements, namely requirements 4 and 5, are removed.
+
+We call $ts_{h}$ the coherence-stabilization time for height $h$.
+
+If we think in terms of $valid(v, bc_{h-1}, x_p)$, notice that it is the application's responsibility
+to ensure 3b, that is, the application designers need to prove that the $x_p$ values at correct processes
+are evolving in a way that eventually `ResponseProcessProposal` returns _Accept_ at all correct processes
+that call `RequestPropocessPropsal` for height $h$ after $ts_{h}$.
+For instance, in Proposer-Based Timestamp, $x_p$ can be considered to be process $p$'s local clock,
+and having clocks synchronized is the mechanism ensuring eventual acceptance of a proposal.
+Finally, it is worth noting that _weak validity_ requires just one correct processes while requirement 3b
+refers to all correct processes.
+
+> [TODO: can we _tighten up_ eventual coherence?... i.e., make it even weaker?]
+>
+> [TODO: Since we have removed determinism, can byzantine proposers cause mayhem now?]
+
+### Modifications to the consensus Algorithm
+
+The Tendermint algorithm as described in the [arXiv paper][arxiv],
+and as implemented in CometBFT up to version `v0.38.0-rc3`,
+cannot guarantee consensus _termination_ for applications
+that just fulfill requirement 3b (eventual coherence), but do not fulfill requirements 3, 4, and 5.
+
+We need the following modifications (in terms of the algorithm as described in page 6 of the arXiv paper):
+
+- remove the evaluation of `valid(v)` in lines 29, 36 and 50 (i.e. replace `valid(v)` by `true`)
+- modify line 23 as follows
+
+> _\[Original\]_ &nbsp; 23: **if** $valid(v) \land (lockedRound_p = −1 \lor lockedValue_p = v)$ **then**
+
+> _\[Modified\]_
+>
+> &nbsp; 23a: $validValMatch := (validRound_p \neq -1 \land validValue_p = v)$
+>
+> &nbsp; 23b: **if** $[lockedRound_p = −1 \land (validValMatch \lor valid(v))] \lor lockedValue_p=v$ **then**
+
+- The occurrences of `valid(v)` that we have removed were in a way redundant,
+  so removing them does not affect the ability of the algorithm to fulfill consensus properties.
```

Comment:
Here you are considering the _weak validity_ property, not the original _validity_ property.

For the latter, if we commit (or act in a way to enable committing) $v$, then we necessarily have to locally verify $v$ by running $valid(v)$.

For the former, the introduced variation, we are removing this method when we have 2f+1 Prevotes for $v$, either explicitly in the same upon clause (lines 29 and 36), or implicitly (line 50).

Reply:
Good point. Reworded the sentence